src/main/java/javaslang/collection/Seq.java

```diff
@@ -198,6 +198,25 @@ default boolean containsSlice(java.lang.Iterable<? extends T> that) {
     }
 
     /**
+     * Tests whether every element of this sequence relates to the corresponding element of another sequence by satisfying a test predicate.
+     *
+     * @param that the other sequence
+     * @param predicate the test predicate, which relates elements from both sequences
+     * @return {@code true} if both sequences have the same length and {@code predicate(x, y)}
+     * is {@code true} for all corresponding elements {@code x} of this sequence and {@code y} of {@code that}, otherwise {@code false}.
+     */
+    default <U> boolean corresponds(java.lang.Iterable<U> that, Predicate<Tuple2<T, U>> predicate) {
```

Comment:
Hey Ruslan,

I urgently need corresponds to fix my local branch. Therefore I will merge the PR now!! I hope this is ok for you.

(I removed Iterator.equals as I explained in #547 and not shouldConvertToMap tests fail. The corresponds will heal that.)